template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerRoleConfigProvider.java

```diff
@@ -74,7 +81,16 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
         return cmTemplateProcessor.isRoleTypePresentInService(getServiceType(), getRoleTypes());
     }
 
-    private Optional<RDSConfig> getFirstRDSConfigOptional(TemplatePreparationObject source) {
-        return source.getRdsConfigs().stream().filter(rds -> DatabaseType.RANGER.name().equalsIgnoreCase(rds.getType())).findFirst();
+    private RDSConfig getRDSConfig(TemplatePreparationObject source) {
+        List<RDSConfig> rdsConfigs = source.getRdsConfigs().stream().
+                filter(rds -> DatabaseType.RANGER.name().equalsIgnoreCase(rds.getType())).collect(Collectors.toList());
+        if (rdsConfigs.size() < 1) {
+            throw new CloudbreakServiceException("Ranger database has not been provided for RANGER_ADMIN component");
+
+        } else if (rdsConfigs.size() > 1) {
+            throw new CloudbreakServiceException("Multiple databases have been provided for RANGER_ADMIN component");
+        }
+
+        return rdsConfigs.get(0);
     }
```

Comment:
If no database is configured, the inbuilt database is used and only one external database of a particular type can be attached, so this check might not be required.

Reply:
Inbuild DB, a DB which is launched by CB is also considered as attached DB in this part of the code.